bindings/rust/s2n-tls-tokio/src/lib.rs

```diff
@@ -311,33 +350,21 @@ where
         Pin::new(&mut tls.stream).poll_flush(ctx)
     }
 
-    fn poll_shutdown(self: Pin<&mut Self>, ctx: &mut Context<'_>) -> Poll<io::Result<()>> {
-        let tls = self.get_mut();
+    fn poll_shutdown(mut self: Pin<&mut Self>, ctx: &mut Context<'_>) -> Poll<io::Result<()>> {
+        ready!(self.as_mut().poll_blinding(ctx))?;
 
-        if tls.blinding.is_none() {
-            let delay = tls
-                .as_ref()
-                .remaining_blinding_delay()
-                .map_err(io::Error::from)?;
-            if !delay.is_zero() {
-                // Sleep operates at the milisecond resolution, so add an extra
-                // millisecond to account for any stray nanoseconds.
-                let safety = Duration::from_millis(1);
-                tls.blinding = Some(Box::pin(sleep(delay.saturating_add(safety))));
-            }
-        };
+        let status = ready!(self.as_mut().with_io(ctx, |mut context| {
```

Comment:
some `poll_*` implementations don't like you polling after it has returned `Ready` (some panic, some don't return the same result) so I think we'll have to store the error somewhere to be safe.